src/AcceptanceTests/StepDefinitions/ApprenticeshipLearnerEventPublishingStepDefinitions.cs

```diff
@@ -0,0 +1,64 @@
+using NServiceBus;
+using SFA.DAS.Apprenticeships.Types;
+using SFA.DAS.NServiceBus.Configuration.NewtonsoftJsonSerializer;
+using QueueNames = SFA.DAS.Funding.ApprenticeshipEarnings.Types.QueueNames;
+
+namespace SFA.DAS.Funding.ApprenticeshipEarnings.AcceptanceTests.StepDefinitions;
+
+[Binding]
+public class ApprenticeshipCreatedEventPublishingStepDefinitions
+{
+    private readonly ScenarioContext _scenarioContext;
+    private static IEndpointInstance? _endpointInstance;
+
+    public ApprenticeshipCreatedEventPublishingStepDefinitions(ScenarioContext scenarioContext)
+    {
+        _scenarioContext = scenarioContext;
+    }
+
+    [BeforeTestRun]
+    public static async Task StartEndpoint()
+    {
+        var endpointConfiguration = new EndpointConfiguration(QueueNames.ApprenticeshipCreated);
+        endpointConfiguration.AssemblyScanner().ThrowExceptions = false;
+        endpointConfiguration.SendOnly();
+        endpointConfiguration.UseNewtonsoftJsonSerializer();
+        endpointConfiguration.Conventions().DefiningEventsAs(x => x == typeof(ApprenticeshipCreatedEvent));
+
+        var transport = endpointConfiguration.UseTransport<LearningTransport>();
+        transport.StorageDirectory(Path.Combine(Directory.GetCurrentDirectory().Substring(0, Directory.GetCurrentDirectory().IndexOf("src")), @"src\.learningtransport"));
+
+        _endpointInstance = await Endpoint.Start(endpointConfiguration)
+            .ConfigureAwait(false);
+    }
+
+    [AfterTestRun]
+    public static async Task StopEndpoint()
+    {
+        await _endpointInstance.Stop()
+            .ConfigureAwait(false);
+    }
+
```

Comment:
can this code be moved into "Hooks" to then be easily shared between different test files. sbus instance can be added to the testcontext

Reply:
Personally I think the logic to setup the endpoint for publishing apprenticeship created events should live in the step file which is scoped to dealing with apprenticeship created events for now, if we want to use this endpoint instance to publish other events in future we can move it at that point. I will however sort this and the one below out so the shared code is in a helper class they both use.